core/src/main/java/org/mskcc/oncotree/utils/CacheUtil.java

```diff
@@ -48,14 +48,23 @@ public static void resetTumorTypesByVersion(Version version) {
         }
         if (tumorTypes.containsKey(version)) {
             logger.debug("getTumorTypesByVersion() -- found '" + version.getVersion() + "' in cache");
-            return tumorTypes.get(version);
+            return getUnmodifiableTumorTypesByVersion(tumorTypes.get(version));
         } else {
             logger.debug("getTumorTypesByVersion() -- did NOT find '" + version.getVersion() + "' in cache, getting now");
             tumorTypes.put(version, TumorTypesUtil.getTumorTypesByVersionFromRaw(version));
-            return tumorTypes.get(version);
+            return getUnmodifiableTumorTypesByVersion(tumorTypes.get(version));
         }
     }
 
+    private static Map<String, TumorType> getUnmodifiableTumorTypesByVersion(Map<String, TumorType> tumorTypeMap) {
+        // code is modifying the returned tumor types, make a copy of everything
+        Map<String, TumorType> unmodifiableTumorTypeMap = new HashMap<String, TumorType>(tumorTypeMap.size());
+        for (Map.Entry<String, TumorType> entry : tumorTypeMap.entrySet()) {
+            unmodifiableTumorTypeMap.put(entry.getKey(), new TumorType(entry.getValue()));
```

Comment:
Ok, I see .. the clone function is for returned values from the CacheUtil. So no duplication within the cache (which uses the more primitive constructors)